### PR TITLE
Fix redirect when deleting a page

### LINKED ISSUE #3654: Delete page shouldn't result in 404
Currently when you delete a page, the user is redirected to the url of the page that was just deleted. It should instead redirect to the parent page, if one exists or to the home page rather than giving the user an error page.

### PR DESCRIPTION
**Pull request by yakky**
Fix #3654 

This works by preserving ``this.options.onClose`` when deleting a page.

This does not fix the redirect when deleting an object in a apphooked application from the frontend

### CODE REVIEW

**Comment by yakky** [comment]
@FinalAngel Could you have a look at this? I've done some manual tests and apparently nothing is broken, but I'm still away from being fully confident

File: cms/static/cms/js/modules/cms.modal.js [hunk 1]
```diff
@@ -584,7 +584,12 @@ $(document).ready(function () {
 				// when the window has been changed pressing the blue or red button, we need to run a reload check
 				// also check that no delete-confirmation is required
 				if(that.saved && !contents.find('.delete-confirmation').length) {
-					that.reloadBrowser(window.location.href, false, true);
+					if(that.options.onClose){
+						that.reloadBrowser(that.options.onClose);
+					}
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.modal.js]: add the else on the same line

File: cms/static/cms/js/modules/cms.modal.js [hunk 2]
```diff
@@ -502,7 +502,7 @@ $(document).ready(function () {
 						if(item.hasClass('default') || item.hasClass('deletelink')) {
  							that.options.newPlugin = null;
  							// reset onClose when delete is triggered
-							if(item.hasClass('deletelink')) that.options.onClose = null;
+							//if(item.hasClass('deletelink')) that.options.onClose = null;
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.modal.js]: space after `//`

**Comment by yakky** [comment]
@mkoistinen This is also present in 3.0.x 
Should I merge this in 3.0.x branch and then forwardport to 3.1?

File: cms/static/cms/js/modules/cms.modal.js [hunk 3]
```diff
@@ -584,7 +582,11 @@ $(document).ready(function () {
 				// when the window has been changed pressing the blue or red button, we need to run a reload check
 				// also check that no delete-confirmation is required
 				if(that.saved && !contents.find('.delete-confirmation').length) {
-					that.reloadBrowser(window.location.href, false, true);
+					if(that.options.onClose){
```
Comment (intellectronica) [inline cms/static/cms/js/modules/cms.modal.js]: ```if (that.options.onClose) {``` (though i see that this format is consistent with what's currently in the file so maybe it's better to change en masse in a later branch).

File: cms/static/cms/js/modules/cms.base.js [hunk 4]
```diff
@@ -53,7 +53,7 @@ $(document).ready(function () {
 						if(response === '' && !url) {
 							// cancel if response is empty
 							return false;
-						} else if(parent.location.pathname !== response) {
+						} else if(response !== '' && parent.location.pathname !== response) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.base.js]: space after `if`

**Comment by vxsx** [comment]
I think js whitespace should be adressed later on, since we are already working on that https://github.com/divio/django-cms/compare/develop...FinalAngel:feature/3.2-javascript

**Comment by FinalAngel** [comment]
LGTM

**Comment by yakky** [comment]
@vxsx @FinalAngel @intellectronica thanks
keep in mind that this is targeted to 3.1, so the style has to be consistent with the same codebase
In must be fixed when ported to develop

@mkoistinen do you want this in 3.0 too?

**Comment by yakky** [comment]
Merged in 3.0.x branch via b7866d7b0cd903282136275be8fd487f645f6983

**Pull request closed without merging**